app/src/main/java/com/moonpig/mvisample/productdetail/ProductDetailViewModel.kt

```diff
@@ -25,30 +25,30 @@ class ProductDetailViewModel(productDetailUseCase: ProductDetailUseCase,
 
     override fun actionFrom(intent: ProductDetailIntent): ProductDetailAction =
             when (intent) {
-                is ProductDetailIntent.Initial -> ProductDetailAction.LoadProductDetail
+                is ProductDetailIntent.Initial -> ProductDetailAction.LoadProductDetail(intent.productId)
                 is ProductDetailIntent.AddToBasket -> ProductDetailAction.AddProductToBasket(intent.productId, intent.quantity)
             }
 
     override fun reduce(previousViewState: ProductDetailScreenViewState, result: ProductDetailResult): ProductDetailScreenViewState =
             when (result) {
                 is ProductDetailResult.GetProductDetail.InFlight -> previousViewState.copy(getProductDetailInFlight = true)
                 is ProductDetailResult.GetProductDetail.Success -> previousViewState.copy(getProductDetailInFlight = false,
-                                                                                                                                     productDetail = ProductDetailViewState(name = result.productDetail.name,
+                                                                                          productDetail = ProductDetailViewState(name = result.productDetail.name,
```

Comment:
Can we be consistent with the naming? Either all should be `getProduct...` or remove the `get` from all of them

Reply:
I don't think changing those names to add/remove get would be good. 

The `getProductDetailInFlight` shows that getting the product details is currently being executed
The `productDetail` is the actual product detail that was resolved earlier. 

But I don't like the `getProductDetailInFlight` in general I think it should be `isFetchingProductDetail` or `isLoadingProductDetail`

Reply:
`getProductDetail` is the name of the method call though, so in this case if it follows the same pattern, it would be `getProductDetailSuccess` or something like that. 

Using `get` was probably a bad word to use as it's confusing with getters and setters. We should probably rename all of them to `fetchProductDetail...InFlight, Success, Error`